### ISSUE TITLE
Zora 0xsplits example?

### ISSUE THREAD

**Issue by rubinovitz** [ourzora/ocs-hackathon-24#2]
Hi I’m trying to create new Zora NFTs with variable revenue splits using an embedded privy wallet and 0xsplits.

do you have examples of your 0xsplits integration I can look at?

Thank you!

**Comment by oveddan** [comment]
hi @rubinovitz are you creating an onchain or premint contract/token?

**Comment by rubinovitz** [comment]
Do you have any documentation on pre-mint? Was unaware it was an option and the one link on Google is broken. Also can't find it in your docs.

**Comment by oveddan** [comment]
hi @rubinovitz we don't currently have much documentation on Premint (we need to add it). but here's some documentation on it from the sdk standpoint:
https://ourzora.github.io/zora-protocol/protocol-sdk/premint-client

to get a better understanding of how it works, you could look at the code for the premint client here:
https://github.com/ourzora/zora-protocol/blob/main/packages/protocol-sdk/src/premint/premint-client.ts#L227

**Comment by rubinovitz** [comment]
@oveddan thanks I think we’re going to do premint

**Comment by oveddan** [comment]
got it - splits on premint are not supported yet (but its planned).

**Comment by rubinovitz** [comment]
@oveddan ok in that case can we do onchain?

**Comment by iainnash** [comment]
Yes it can be done onchain – you can submit the split on chain before a premint or submit both the mint and the split onchain. You can just use the splits SDK/API and then submit that address to the contract payoutaddress.

**Comment by oveddan** [comment]
we should have this supported in our sdk soon

**Comment by oveddan** [comment]
hey we have an updated sdk and example for creating with splits, check it out here:
https://ourzora.github.io/zora-protocol/protocol-sdk/creator/splits

**Comment by rubinovitz** [comment]
@oveddan is your documentation for contract metadata correct? If so what is the difference between image and imageURI?

**Comment by mtple** [comment]
@oveddan The collection metadata here https://docs.zora.co/docs/smart-contracts/creator-tools/Deploy1155Contract is what @rubinovitz is referring to. 

I am also confused about how these two pages of documentation work together because there is also this one https://ourzora.github.io/zora-protocol/protocol-sdk/creator/onchain

**Comment by oveddan** [comment]
hey weve been migrating our docs so there is a bit of confusion/duplication between the two.

For now you can know that:


Contract metadata has the following json schema:

```json twoslash
{
  "name": "{contract name}",
  "description": "{contract description}",
  "image": "{contract image}"
} 
```

Token metadata has the following json schema:

```json twoslash
{
  "name": "{token name}",
  "description": "{token description}",
  "image": "{token image ipfs url}",
  "animation_url": "{token media ipfs url}",
  "content": {
    "mime": "{content mime type}",
    "uri": "{content ipfs uri}"
  },
  "attributes": {
    "trait_type": "{trait type}",
    "value": "{trait value}"
  }
}
```

Contract and token metadata should be pinned to IPFS using a preferred pinning service, 
and the url when set on a contract or token should be in an ipfs url format, like: `ipfs://{cid}`

**Comment by rubinovitz** [comment]
Thanks!

What is the data type of the contract image?

**Comment by mtple** [comment]
![Screenshot 2024-07-08 at 8 32 39 AM](https://github.com/ourzora/ocs-hackathon-24/assets/73853716/b9d0a935-abf2-4d1c-91cf-36710d673659)
@oveddan It appears that all fields in the contract metadata are optional. Does that mean that this json file itself is optional for creating an erc1155 collection?

Clearly, there needs to be a contract to create a token, but I am confused as to why all fields in the contract metadata would be optional.
